src/intensity_filter.cpp

```diff
@@ -0,0 +1,112 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2008, Willow Garage, Inc.
+*  Copyright (c) 2020, Eurotec B.V.
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Willow Garage nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*
+*  \author Vijay Pradeep, Rein Appeldoorn
+*
+*********************************************************************/
+
+#include <laser_filters/intensity_filter.h>
+#include <ros/node_handle.h>
+
+namespace laser_filters
+{
+LaserScanIntensityFilter::LaserScanIntensityFilter()
+{
+}
+
+bool LaserScanIntensityFilter::configure()
+{
+  ros::NodeHandle private_nh("~" + getName());
+  dyn_server_.reset(new dynamic_reconfigure::Server<IntensityFilterConfig>(own_mutex_, private_nh));
+  dynamic_reconfigure::Server<IntensityFilterConfig>::CallbackType f;
+  f = boost::bind(&LaserScanIntensityFilter::reconfigureCB, this, _1, _2);
+  dyn_server_->setCallback(f);
+
+  getParam("lower_threshold", config_.lower_threshold);
+  getParam("upper_threshold", config_.upper_threshold);
+  getParam("invert", config_.invert);
+
+  getParam("filter_override_range", config_.filter_override_range);
+  getParam("filter_override_intensity", config_.filter_override_intensity);
```

Comment:
Doesn't dynamic_reconfigure read the ROS parameters and call your callback for you? Do you really need to manually read all the variables and call updateConfig here?

Reply:
That is correct but since we would like to be backwards compatible with the filter chain parameter implementation, we have to add these `getParam` calls in order to ensure that old configs still work. Parameters are read from the `params` namespace in the defined filter array item.

Reply:
Got it, thanks!